MTA_limit_trends/Comp/run_comp_grad_data_update.py

```diff
@@ -618,49 +640,37 @@ def find_limits(stime, mkey, alimit):
-    output: ./<msid>_<dtype>_data.fits
+    output: ./<msid>_<dtype>_data table HDU
     """
-    cols    = col_names
+    cols    = COL_NAMES
     cols[1] = msid
 
-    c1  = Column(name=cols[0],  format=col_format[0],  array = data[0])
-    c2  = Column(name=cols[1],  format=col_format[1],  array = data[1])
-    c3  = Column(name=cols[2],  format=col_format[2],  array = data[2])
-    c4  = Column(name=cols[3],  format=col_format[3],  array = data[3])
-    c5  = Column(name=cols[4],  format=col_format[4],  array = data[4])
-    c6  = Column(name=cols[5],  format=col_format[5],  array = data[5])
-    c7  = Column(name=cols[6],  format=col_format[6],  array = data[6])
-    c8  = Column(name=cols[7],  format=col_format[7],  array = data[7])
-    c9  = Column(name=cols[8],  format=col_format[8],  array = data[8])
-    c10 = Column(name=cols[9],  format=col_format[9],  array = data[9])
-    c11 = Column(name=cols[10], format=col_format[10], array = data[10])
-    c12 = Column(name=cols[11], format=col_format[11], array = data[11])
-    c13 = Column(name=cols[12], format=col_format[12], array = data[12])
-    c14 = Column(name=cols[13], format=col_format[13], array = data[13])
-    c15 = Column(name=cols[14], format=col_format[14], array = data[14])
-    c16 = Column(name=cols[15], format=col_format[15], array = data[15])
+    c1  = Column(name=cols[0],  format=COL_FORMAT[0],  array = data[0])
+    c2  = Column(name=cols[1],  format=COL_FORMAT[1],  array = data[1])
+    c3  = Column(name=cols[2],  format=COL_FORMAT[2],  array = data[2])
+    c4  = Column(name=cols[3],  format=COL_FORMAT[3],  array = data[3])
+    c5  = Column(name=cols[4],  format=COL_FORMAT[4],  array = data[4])
+    c6  = Column(name=cols[5],  format=COL_FORMAT[5],  array = data[5])
+    c7  = Column(name=cols[6],  format=COL_FORMAT[6],  array = data[6])
+    c8  = Column(name=cols[7],  format=COL_FORMAT[7],  array = data[7])
+    c9  = Column(name=cols[8],  format=COL_FORMAT[8],  array = data[8])
+    c10 = Column(name=cols[9],  format=COL_FORMAT[9],  array = data[9])
+    c11 = Column(name=cols[10], format=COL_FORMAT[10], array = data[10])
+    c12 = Column(name=cols[11], format=COL_FORMAT[11], array = data[11])
+    c13 = Column(name=cols[12], format=COL_FORMAT[12], array = data[12])
+    c14 = Column(name=cols[13], format=COL_FORMAT[13], array = data[13])
+    c15 = Column(name=cols[14], format=COL_FORMAT[14], array = data[14])
+    c16 = Column(name=cols[15], format=COL_FORMAT[15], array = data[15])
         
     coldefs = pyfits.ColDefs([c1, c2, c3, c4, c5, c6, c7, c8, c9, c10, c11, c12, c13, c14, c15, c16])
     tbhdu   = pyfits.BinTableHDU.from_columns(coldefs)
 
-    if dtype == 'week':
-        ofits = msid + '_week_data.fits'
-    elif dtype == 'short':
-        ofits = msid + '_short_data.fits'
-    else:
-        ofits = msid + '_data.fits'
-    
-    mcf.rm_files(ofits)
-
-    tbhdu.writeto(ofits)
-
-    return ofits
+    return tbhdu
```

Comment:
The script is currently formatted to slice the full data fits files into smaller time selections for plotting based on the current day the script is running. For example if it is day 130 of 2024, then the smaller time fits files will all end at the end of day 128 by design.

This operates normally under our approach of running the cron job once a day. However, when running this script multiple times in one day (as in a testing scenario), the script will generate an appending table which overwrites the original table due to using the same naming scheme. Instead, the file writing is taken outside of this function so that the process of appending a new fits file can be more easily controlled.